src/blockchain/fork_choice.cpp

```diff
@@ -482,6 +483,64 @@ namespace lean {
     return true;
   }
 
+  void ForkChoiceStore::updateLastFinalized(const Checkpoint& checkpoint) {
+    BOOST_ASSERT(checkpoint.slot > latest_finalized_.slot);
+    latest_finalized_ = checkpoint;
+
+    pruneStatesAfterFinalized();
+
+    // Safety: pull up head/safe_target up to the finalized root (just in case).
+    if (not blocks_.contains(head_)) {
+      head_ = latest_finalized_.root;
```

Comment:
Can finalized block have children? Should choose head from them?
Can use `updateHead`?